.github/workflows/ci.yaml

```diff
@@ -0,0 +1,19 @@
+name: CI
+
+on:
+  workflow_call:
+
+jobs:
+  unit-test:
```

Comment:
Yes, the reason the Python installation action is used here is to guarantee the Python version being used, otherwise the workflow is going to use whatever is available in ubuntu-latest by default (which was 3.8 in this case which caused the tests to fail due to using some Python 3.10 features in this charm around type hinting. I'm happy to propose a pull request for the indigo charm to use the Python action there as well

Reply:
The `charmcraft.yaml` has "build on" and "run on" 20.04. Will that cause any issues here if we're depending on 3.10 features?

Reply:
Yes at it turns out when I started the integration tests, I have since updated that to 22.04 and that'll be part of the PR for the integration tests